rasa/core/actions/forms.py

```diff
@@ -437,7 +437,9 @@ def _temporary_tracker(
             current_tracker.events_after_latest_restart()
             # Insert form execution event so that it's clearly distinguishable which
             # events were newly added.
-            + [ActionExecuted(self.name())] + additional_events,
+            + [SlotSet(REQUESTED_SLOT, self.get_slot_to_fill(current_tracker))]
+            + [ActionExecuted(self.name())]
```

Comment:
```suggestion
            # Insert form execution event so that it's clearly distinguishable which
            # events were newly added.
            + [ActionExecuted(self.name())]
```